src/ui/components/Breethe/component.ts

```diff
@@ -128,7 +128,8 @@ export default class Breethe extends Component {
         params = params as SearchParams;
         this.locationId = null;
         this.searchTerm = params.searchTerm;
-        this.coordinates = params.coordinates;
+        let { coordinates } = params;
+        this.coordinates = (coordinates) ? coordinates.match(/[-+]?[0-9]*\.?[0-9]+/g) : null;
```

Comment:
Is this really necessary? If coordinates wouldn't actually match [here](https://github.com/simplabs/breethe-client/pull/92/files#diff-a0d31ced42dfb7f0cc15c63975bb82e8R117), they should be `null` anyway. I'd probably capture `lan` and `lon` separately in the router and pass them as an array into `_setMode`. That should be much simpler?